### PR TITLE
Enable -Wall, -Wuninitialized, -Wunreachable-code for building runtime code.

### PR DESCRIPTION
**Pull request by tkremenek**

### CODE REVIEW

**Comment by tkremenek** [comment]
@swift-ci test

**Comment by jckarter** [comment]
Thanks Ted! We should make sure to tag the `RemoteAST`, `Reflection` and `RemoteMirror` libraries with these flags too.

**Comment by gparker42** [comment]
Is #14417 a better place to change this? Does that cover the other libraries that Joe mentioned?

File: stdlib/public/CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,24 @@ if(CMAKE_SYSTEM_NAME STREQUAL "CYGWIN")
...
+endif()
+
+# Build C++ runtime code with -Wunreachable-code
+check_cxx_compiler_flag("-Wunreachable-code" CXX_SUPPORTS_UNREACHABLE_WARNING)
+if(CXX_SUPPORTS_UNREACHABLE_WARNING)
+  list(APPEND SWIFT_RUNTIME_CORE_CXX_FLAGS "-Wunreachable-code")
+endif()
+
+# Build C++ runtime code with -Wuninitialized
+check_cxx_compiler_flag("-Wunreachable-code" CXX_SUPPORTS_UNINIT_WARNING)
```
Comment (dcci) [inline stdlib/public/CMakeLists.txt]: Should this be `-Wuninitialized`?

File: stdlib/public/CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,24 @@ if(CMAKE_SYSTEM_NAME STREQUAL "CYGWIN")
...
+if(CXX_SUPPORTS_UNINIT_WARNING)
+  list(APPEND SWIFT_RUNTIME_CORE_CXX_FLAGS "-Wuninitialized")
```
Comment (dcci) [inline stdlib/public/CMakeLists.txt]: I think that `-Wall` implies `-Wuninitialized` at least for clang and gcc.

**Review by gparker42 — CHANGES_REQUESTED** [review]
On hold. We want at most one of #14415 and #14417.

**Comment by tkremenek** [comment]
I think #14417 is a better place for this.  However, I'd like to do an audit of what warnings are being enabled for various source files to see if anything is being missed.  I'll do that as a separate investigation.

**Pull request closed without merging**